testsuite/classlibrary/TestDocument.sc

```diff
@@ -0,0 +1,53 @@
+TestDocument : UnitTest {
+	// autorun doesn't have access to this object's scope
+	// need a public place to communicate the result
+	classvar <>success;
+
+	test_new_document_runs_initAction {
+		var success = false, doc, save;
+		if(Platform.ideName == "scqt") {
+			save = Document.initAction;
+			protect {
+				Document.initAction = { success = true };
+				doc = Document.new;
+				0.1.wait;
+				doc.close;
+			} {
+				Document.initAction = save;
+			};
+			this.assert(success, "Document.new should fire Document.initAction function");
+		} {
+			this.assert(false, "Cannot test Document.new behavior in a non-ScIDE session");
+		};
+	}
+
+	test_open_autorun_document_runs_code {
+		var path = PathName.tmp +/+ "autoRunTest%.scd".format(UniqueID.next),
+		file, doc;
+		if(Platform.ideName == "scqt") {
+			if(File.exists(path)) {
+				this.assert(false, "Document autoRun test file path should not exist: '%'".format(path));
```

Comment:
would raising an exception make more sense here than `assert false`? it's not the thing we're really testing here.

Reply:
It was never clear to me what is the outcome of raising an exception in a test. What eventually happens upstream? I guess it makes sense if the test is reported as a failure due to exception.

Reply:
i think the entire test suite exits. man that's annoying. i'd expect an exception to be pretty much the same as a failed assertion. no need for you to change anything here then.